compiler/luci-interpreter/src/kernels/ArgMax.cpp

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "kernels/ArgMax.h"
+#include "kernels/Utils.h"
+#include <tensorflow/lite/kernels/internal/optimized/optimized_ops.h>
+
+namespace luci_interpreter
+{
+namespace kernels
+{
+
+ArgMax::ArgMax(const Tensor *input, const Tensor *axis, Tensor *output, const ArgMaxParams &params)
+    : KernelWithParams<ArgMaxParams>(params), _input(input), _axis(axis), _output(output)
+{
+}
+
+void ArgMax::configure()
+{
+  assert(_axis->shape().num_elements() == 1);
+  assert(_axis->element_type() == DataType::S32 || _axis->element_type() == DataType::S64);
+  assert(_input->shape().num_dims() >= 1);
+  const Shape &input_shape = _input->shape();
+  const int num_dims = input_shape.num_dims();
+  Shape output_shape(num_dims - 1);
+  const auto *axis = getTensorData<int32_t>(_axis);
+  int j = 0;
+  for (int i = 0; i < num_dims; i++)
+  {
+    if (i == axis[0])
+      continue;
+    output_shape.dim(j++) = input_shape.dim(i);
+  }
+  // TODO output type setting needed.
```

Comment:
Output type would have been set already. TFLite sets the output type in `Prepare`, but luci-interpreter sets the type of all output tensors (except CircleOutput) in `createTensors`. 

@s-barannikov , Do we need to reset the type here?

Reply:
AFAIK You are right about redundancy of the setting.
I think we can just `assert` that the types match.